runtime/commonMain/src/kotlinx/serialization/SerialImplicits.kt

```diff
@@ -34,43 +32,28 @@ public annotation class ImplicitReflectionSerializer
 @RequiresOptIn(level = WARNING)
 public annotation class UnstableDefault
 
-@ImplicitReflectionSerializer
-public inline fun <reified T : Any> BinaryFormat.dump(value: T): ByteArray =
-    dump(context.getContextualOrDefault(T::class), value)
+public inline fun <reified T : Any> BinaryFormat.dump(value: T): ByteArray = dump(context.getContextualOrDefault(), value)
 
-@ImplicitReflectionSerializer
-public inline fun <reified T : Any> BinaryFormat.dumps(value: T): String =
-    dumps(context.getContextualOrDefault(T::class), value)
+public inline fun <reified T : Any> BinaryFormat.dumps(value: T): String = dumps(context.getContextualOrDefault(), value)
 
-@ImplicitReflectionSerializer
-public inline fun <reified T : Any> BinaryFormat.load(raw: ByteArray): T =
-    load(context.getContextualOrDefault(T::class), raw)
+public inline fun <reified T : Any> BinaryFormat.load(raw: ByteArray): T = load(context.getContextualOrDefault(), raw)
 
-@ImplicitReflectionSerializer
-public inline fun <reified T : Any> BinaryFormat.loads(hex: String): T =
-    loads(context.getContextualOrDefault(T::class), hex)
+public inline fun <reified T : Any> BinaryFormat.loads(hex: String): T = loads(context.getContextualOrDefault(), hex)
 
-
-@ImplicitReflectionSerializer
 public inline fun <reified T : Any> StringFormat.stringify(value: T): String =
-    stringify(context.getContextualOrDefault(T::class), value)
+    stringify(context.getContextualOrDefault(), value)
 
-@ImplicitReflectionSerializer
 public inline fun <reified T : Any> StringFormat.stringify(objects: List<T>): String =
```

Comment:
Btw, all these List and Map overloads were added because of incorrect work with generics. Since KType API does not have this problem, they _probably_can be deprecated.